### PR TITLE
Palemoon 28.3.0 (from binary tarball)

### PR DESCRIPTION
**Pull request by DevElCuy**
###### Motivation for this change

Easy install palemoon 28.3.0 (currently the latest release), which includes security updates. The build from sources is currently broken (see #51161).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Infinisil** [comment]
Have you tested the binary? Because it doesn't work for me.

```
$ nix-build -A palemoon-bin
/nix/store/01638y9a4naccywvn05sml0l11ryd9c4-palemoon-bin-28.2.2
$ result/bin/palemoon
result/bin/palemoon: command not found
$ ldd result/bin/palemoon
  linux-vdso.so.1 (0x00007ffe98cab000)
  libpthread.so.0 => /nix/store/1mnsmslnx5anjfksac6417xfzzglrwhr-glibc-2.27/lib/libpthread.so.0 (0x00007f6b5711f000)
  libdl.so.2 => /nix/store/1mnsmslnx5anjfksac6417xfzzglrwhr-glibc-2.27/lib/libdl.so.2 (0x00007f6b56f1b000)
  libstdc++.so.6 => not found
  libm.so.6 => /nix/store/1mnsmslnx5anjfksac6417xfzzglrwhr-glibc-2.27/lib/libm.so.6 (0x00007f6b56b86000)
  libgcc_s.so.1 => /nix/store/1mnsmslnx5anjfksac6417xfzzglrwhr-glibc-2.27/lib/libgcc_s.so.1 (0x00007f6b56970000)
  libc.so.6 => /nix/store/1mnsmslnx5anjfksac6417xfzzglrwhr-glibc-2.27/lib/libc.so.6 (0x00007f6b565bc000)
  /lib64/ld-linux-x86-64.so.2 => /nix/store/1mnsmslnx5anjfksac6417xfzzglrwhr-glibc-2.27/lib64/ld-linux-x86-64.so.2 (0x00007f6b5733e000)
```

I'm not sure myself how `libstdc++` could be fixed though.

**Comment by worldofpeace** [comment]
Maybe try `autoPatchelfHook` since I don't see the binary being patched anywhere in the derivation?

Also why the `orig` output?

File: pkgs/applications/networking/browsers/palemoon-bin/default.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    url = "https://linux.palemoon.org/datastore/release/palemoon-${version}.${arch}.tar.bz2";
+    sha256 = "f1a6a54238a658d645149736e21074c51a1c770ad0b084500b8489f807eb700d";
+  };
+  outputs = [ "out" "orig" ];
+  sourceRoot = ".";
+  unpackCmd = ''
+    cat "$src" | tar xj
+  '';
+
+  buildPhase = ":";   # nothing to build
```
Comment (worldofpeace) [inline pkgs/applications/networking/browsers/palemoon-bin/default.nix]: Can't we do `dontBuild = true;`?

**Comment by worldofpeace** [comment]
> Binaries get patched already during build

Unfortunately this is not that simple, though I wish it was.

I recommend giving https://nixos.wiki/wiki/Packaging/Binaries a read and possibly anything to do with patching binaries like this under nix.

**Comment by worldofpeace** [comment]
@develCuy I've noticed that you've been updating this but you haven't made the requested changes.

These binaries won't work because they aren't patched.
We cannot expect a binary package like that in nixpkgs.

Please see my comment at https://github.com/NixOS/nixpkgs/pull/52728#issuecomment-450586510 or ask in #nixos on freenode if you need further guidance.

**Comment by worldofpeace** [comment]
Palemoon has been updated from source #64499

**Pull request closed without merging**